Makefile

```diff
@@ -95,64 +107,84 @@ test-unit-go:
 # Build / Publish                                                              #
 ################################################################################
 
-.PHONY: build
-build: build-images xbuild-cli
-
-.PHONY: build-images
-build-images: build-apiserver build-scheduler build-observer
-
 .PHONY: build-logger-linux
 build-logger-linux:
-	docker build \
-		-t $(DOCKER_IMAGE_PREFIX)brigade-logger-linux:$(IMMUTABLE_DOCKER_TAG) \
-		- < v2/logger/Dockerfile.linux
-	docker tag $(DOCKER_IMAGE_PREFIX)brigade-logger-linux:$(IMMUTABLE_DOCKER_TAG) $(DOCKER_IMAGE_PREFIX)brigade-logger-linux:$(MUTABLE_DOCKER_TAG)
+	$(KANIKO_DOCKER_CMD) /kaniko/executor \
+		--dockerfile /workspaces/brigade/v2/logger/Dockerfile.linux \
+		--context dir:///workspaces/brigade/logger \
+		--no-push
 
 .PHONY: build-%
 build-%:
-	docker build \
-		-f v2/$*/Dockerfile \
-		-t $(DOCKER_IMAGE_PREFIX)brigade-$*:$(IMMUTABLE_DOCKER_TAG) \
-		--build-arg VERSION='$(VERSION)' \
-		--build-arg COMMIT='$(GIT_VERSION)' \
-		.
-	docker tag $(DOCKER_IMAGE_PREFIX)brigade-$*:$(IMMUTABLE_DOCKER_TAG) $(DOCKER_IMAGE_PREFIX)brigade-$*:$(MUTABLE_DOCKER_TAG)
+	$(KANIKO_DOCKER_CMD) /kaniko/executor \
+		--dockerfile /workspaces/brigade/v2/$*/Dockerfile \
+		--context dir:///workspaces/brigade/ \
+		--no-push
+
+.PHONY: push-logger-linux
+push-logger-linux:
```

Comment:
These push targets appear to re-build first.  Is that as intended (and/or how kaniko works)?

Reply:
It's just how it works. Container comes up builds + pushes (unless you tell it not to push) and then goes down. When it goes down, the image you built within it is gone. So with the images we build being so ephemeral, build + push necessarily happen in one fell swoop.